aqueduct/gateway/views/decorators.py

```diff
@@ -465,3 +610,57 @@ async def wrapper(request: ASGIRequest, *args, **kwargs):
         return await view_func(request, *args, **kwargs)
 
     return wrapper
+
+
+def parse_jsonrpc_message(view_func):
+    @wraps(view_func)
+    async def wrapper(request: ASGIRequest, *args, **kwargs):
+        session_id = request.headers.get("Mcp-Session-Id")
+        kwargs["session_id"] = session_id
+
+        if request.method != "POST":
+            return await view_func(request, request_log=None, *args, **kwargs)
+
+        try:
+            body = request.body
+            logger.error(f"Request body in parse_jsonrpc_message: {body!r}")
+            if body is None:
+                logger.error("Request body is None")
+                return JsonResponse({"error": "Missing request body"}, status=400)
+            data = json.loads(body)
+            json_rpc_message = None
+
+            try:
+                from mcp.types import JSONRPCMessage
```

Comment:
Unless there is a particular reason to import it here, I would suggest moving it to the top of the module, where all other imports are (that's also what [PEP8](https://pep8.org/#imports) recommends)